grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details.ng.html

```diff
@@ -6,26 +6,38 @@
 </div>
 
 <ng-template #noError>
-  <div *ngIf="flowRunning$ | async" class="in-progress">
+  <div *ngIf="flowRunning$ | async; else showResults" class="in-progress">
     Query in progress: {{ (args$ | async).query }}
   </div>
 
-  <div *ngIf="resultsTable$ | async as resultsTable; else noResultsTable" class="results-table">
-    <osquery-results-table
-      [tableToDisplay]="resultsTable">
-    </osquery-results-table>
-  </div>
-
-  <ng-template #noResultsTable>
-    <div *ngIf="progressTable$ | async as progressTable" class="progress-table">
-      <div *ngIf="additionalRowsAvailable$ | async as additionalRowsAvailable" class="show-additional">
-        <span class="show-additional-text">{{additionalRowsAvailable}} row(s) hidden.</span>
-        <button (click)="fullTableRequested()">Display the full table</button>
-      </div>
-
+  <ng-template #showResults>
+    <div *ngIf="resultsTable$ | async as resultsTable; else noResultsTable" class="results-table">
       <osquery-results-table
-        [tableToDisplay]="progressTable">
+        [table]="resultsTable">
       </osquery-results-table>
     </div>
+
+    <ng-template #noResultsTable>
+      <div *ngIf="progressTable$ | async as progressTable; else noTables" class="progress-table">
+        <osquery-results-table
+          [table]="progressTable">
+        </osquery-results-table>
+
+        <ng-container *ngIf="(additionalRowsAvailable$ | async) !== 0">
+          <div
+          *ngIf="additionalRowsAvailable$ | async as additionalCount; else progressCountUndefined"
```

Comment:
This ngIf is duplicated and looks a bit like code smell. Maybe you can change additionalRowsAvailable$ to never be undefined?

Reply:
I'm actually using the undefined value - the case that I try to cover is if for some reason `totalRowCount` in the progress is undefined, and the values in `additionalRowsAvailable$` are too, in which case something went wrong and we still want to display the "View all" button.

I'm not sure what would be a cleaner approach.

Reply:
I suggest simplifying the code: Remove #progressCountUndefined to re-use one `button` for both cases. This button can then show `({{additionalCount}} more)` if they exist or hide the "more" count, if 0 or undefined.

From a users point of view "Results might be missing or incomplete" does not provide any new actionable information.